File: LeetCode/TwoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 * 
 * Problem solved by Daniel Shwan
 * Runtime: O(n)
 */
var twoSum = function(nums, target) {
    var i, j;

    for(i = 0; i < nums.length; i++){
        for(j = i+1; j < nums.length; j++){
            if(nums[i]+nums[j] == target){
                return [i,j];
            }
        }
    }
};

File: LeetCode/Python/TwoSum.py
'''
Improved solution compared to the one done in JS
'''
class Solution:
   def twoSum(self, nums: List[int], target: int) -> List[int]:
        
      dict = {}
        
      for i in range(len(nums)):
         dict[nums[i]] = i
        
      for j in range(len(nums)):
            
         temp = target - nums[j]     
         if dict.get(temp) and nums[dict[temp]] + nums[j] == target:
                
            if dict[temp] != j: return [j, dict[temp]]